### ISSUE TITLE
Eiger configuration

### ISSUE THREAD

**Issue by mkotiuga** [aiidalab/aiidalab-qe#105]
The configuration seems to be a bit off: (1) the OMP_NUM_THREADS isn't getting the correct slurm env var; (2) the wall-time for the debug queue is set twice; (3) QE start running on 256 nodes?

(1)
The prepend text 
`OMP_NUM_THREADS=${SLURM_CPUS_PER_TASK}`

results in an error for each job I run 

e.g for pk 534 in _scheduler-stderr.txt:

```
OMP: Warning #232: OMP_NUM_THREADS: Invalid symbols found. Check the value "".
_scheduler-stderr.txt lines 1-9/9 (END)

```
(2) For the Eiger-debug config, the wall time is set twice (doesn't throw any error, but its a bit odd):
 pk 534 _aiidasubmit.sh

```
#!/bin/bash
#SBATCH --no-requeue
#SBATCH --job-name="aiida-534"
#SBATCH --get-user-env
#SBATCH --output=_scheduler-stdout.txt
#SBATCH --error=_scheduler-stderr.txt
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --time=12:00:00


### computer prepend_text start ###
#SBATCH --partition=debug
#SBATCH --account=mr0
#SBATCH --constraint=mc
#SBATCH --time=00:30:00

export OMP_NUM_THREADS=${SLURM_CPUS_PER_TASK}
export OMP_PROC_BIND=close
export OMP_PLACES=cores
### computer prepend_text end ###
module load cpeIntel/21.06
module load QuantumESPRESSO/6.7.0


'srun' '-n' '1' '/apps/eiger/UES/jenkins/1.4.0/software/QuantumESPRESSO/6.7.0-cpeIntel-21.06/bin/pw.x' '-in' 'aiida.in'  >'aiida.out'

```
(3) QE starts to run on 256 cores? (aren't there 128 on each eiger node?) - Maybe this is an issue with their QE module not the configuration of the computer/code in aiidaLab(?)
pk 534 header of aiida.out

    ```
 Program PWSCF v.6.7MaX starts on 27Jul2021 at 12:11:10

     This program is part of the open-source Quantum ESPRESSO suite
     for quantum simulation of materials; please cite
         "P. Giannozzi et al., J. Phys.:Condens. Matter 21 395502 (2009);
         "P. Giannozzi et al., J. Phys.:Condens. Matter 29 465901 (2017);
          URL http://www.quantum-espresso.org",
     in publications or presentations arising from this work. More details at
     http://www.quantum-espresso.org/quote

     Parallel version (MPI & OpenMP), running on     256 processor cores
     Number of MPI processes:                 1
     Threads/MPI process:                   256

     MPI processes distributed on     1 nodes
     Reading input from aiida.in
```

**Comment by csadorf** [comment]
@mbercx Is the walltime set by the workflow or so?

**Comment by csadorf** [comment]
> The configuration seems to be a bit off: (1) the OMP_NUM_THREADS isn't getting the correct slurm env var; (2) the wall-time for the debug queue is set twice; (3) QE start running on 256 nodes?
> 
Thanks a lot for the detailed report!

> (1)
> The prepend text
> `OMP_NUM_THREADS=${SLURM_CPUS_PER_TASK}`
> 
> results in an error for each job I run
> 
> e.g for pk 534 in _scheduler-stderr.txt:
> 
> ```
> OMP: Warning #232: OMP_NUM_THREADS: Invalid symbols found. Check the value "".
> _scheduler-stderr.txt lines 1-9/9 (END)
> ```
I guess we would need to set the `#SBATCH --cpus-per-task=X` directive for the `${SLURM_CPUS_PER_TASK}` variable to be set (see the example script on https://confluence.cscs.ch/pages/viewpage.action?pageId=284426490 ).
> 
> (2) For the Eiger-debug config, the wall time is set twice (doesn't throw any error, but its a bit odd):
>
> <details>
> <summary>pk 534 _aiidasubmit.sh</summary>
>
> ```
> #!/bin/bash
> #SBATCH --no-requeue
> #SBATCH --job-name="aiida-534"
> #SBATCH --get-user-env
> #SBATCH --output=_scheduler-stdout.txt
> #SBATCH --error=_scheduler-stderr.txt
> #SBATCH --nodes=1
> #SBATCH --ntasks-per-node=1
> #SBATCH --time=12:00:00
> 
> 
> ### computer prepend_text start ###
> #SBATCH --partition=debug
> #SBATCH --account=mr0
> #SBATCH --constraint=mc
> #SBATCH --time=00:30:00
> 
> export OMP_NUM_THREADS=${SLURM_CPUS_PER_TASK}
> export OMP_PROC_BIND=close
> export OMP_PLACES=cores
> ### computer prepend_text end ###
> module load cpeIntel/21.06
> module load QuantumESPRESSO/6.7.0
> 
> 
> 'srun' '-n' '1' '/apps/eiger/UES/jenkins/1.4.0/software/QuantumESPRESSO/6.7.0-cpeIntel-21.06/bin/pw.x' '-in' 'aiida.in'  >'aiida.out'
> ```
> 
> </details>

Indeed, I assume that the walltime is set somewhere else already (maybe the workflow @mbercx )? At least setting it again, overrides the previous setting. While odd, this might be the "best" way to ensure that jobs to the debug partition are set to 30 min of walltime no matter what is specified elsewhere.

> (3) QE starts to run on 256 cores? (aren't there 128 on each eiger node?) - Maybe this is an issue with their QE module not the configuration of the computer/code in aiidaLab(?)
>
> <details>
> <summary>pk 534 header of aiida.out</summary>
>
> 
> ```
> 
> Program PWSCF v.6.7MaX starts on 27Jul2021 at 12:11:10
> 
>  This program is part of the open-source Quantum ESPRESSO suite
>  for quantum simulation of materials; please cite
>      "P. Giannozzi et al., J. Phys.:Condens. Matter 21 395502 (2009);
>      "P. Giannozzi et al., J. Phys.:Condens. Matter 29 465901 (2017);
>       URL http://www.quantum-espresso.org",
>  in publications or presentations arising from this work. More details at
>  http://www.quantum-espresso.org/quote
> 
>  Parallel version (MPI & OpenMP), running on     256 processor cores
>  Number of MPI processes:                 1
>  Threads/MPI process:                   256
> 
>  MPI processes distributed on     1 nodes
>  Reading input from aiida.in
> ```
> </details>

Looks like hyperthreading is active. Each eiger node has two sockets with 64 cores that can be hyper threaded to 128 threads. I am not sure how to deactivate that.

**Comment by mbercx** [comment]
> Indeed, I assume that the walltime is set somewhere else already (maybe the workflow @mbercx )?

Yes, the protocol has a default walltime of 12 hours, that is set on the inputs of the `PwCalculation`.

**Comment by csadorf** [comment]
> > Indeed, I assume that the walltime is set somewhere else already (maybe the workflow @mbercx )?
> 
> Yes, the protocol has a default walltime of 12 hours, that is set on the inputs of the `PwCalculation`.

So, is there a better option to deal with overwriting that for specific codes as in the case of "pw@eiger-debug"?

**Comment by mbercx** [comment]
> So, is there a better option to deal with overwriting that for specific codes as in the case of "pw@eiger-debug"?

Hmm, I think using the prepend text makes sense, since it's specifically for this code? You could also pop the input on the `PwCalculation` when this code is run, but that just adds extra code to maintain and the only benefit is that the submission script is cleaner, right? The correct wall time is the last one specified?

**Comment by csadorf** [comment]
> > So, is there a better option to deal with overwriting that for specific codes as in the case of "pw@eiger-debug"?
> 
> Hmm, I think using the prepend text makes sense, since it's specifically for this code? You could also pop the input on the `PwCalculation` when this code is run, but that just adds extra code to maintain and the only benefit is that the submission script is cleaner, right? The correct wall time is the last one specified?

Yes, exactly. That's why I figured that this approach might actually be the "best" one even if it is a bit odd that the walltime is specified multiple times.
